chat/views/get_message_count_by_bot.py

```diff
@@ -0,0 +1,123 @@
+    operation_id="GetMessageCountAggregateByBot",
+    operation_summary="Get message counts sent to a bot over time",
+    manual_parameters=[
+        openapi.Parameter(
+            name='sort',
+            in_=openapi.IN_QUERY,
+            type=openapi.TYPE_STRING,
+            description='Date sorting order for the results (asc or desc), default is descending',
+            required=False
+        ),
+        openapi.Parameter(
+            name='daysLimit',
+            in_=openapi.IN_QUERY,
+            type=openapi.TYPE_INTEGER,
+            description='Number of days to limit the data to, e.g., the last 30 days',
+            required=False
+        ),
+        openapi.Parameter(
+            name='botId',
+            in_=openapi.IN_QUERY,
+            type=openapi.TYPE_STRING,
+            description='ID of the bot to filter messages by (required)',
+            required=True
+        ),
+    ],
+    responses={
+        "200": openapi.Response(
+            description="Messages retrieved successfully",
+            examples={
+                "application/json": {
+                    "timestamp": "2021-08-30 14:00:00",
+                    "status": 200,
+                    "msg": "Messages retrieved successfully",
+                    "data": []
+                }
+            }
+        ),
+        "400": openapi.Response(
+            description="Bad request",
+            examples={
+                "application/json": {
+                    "timestamp": "2021-08-30 14:00:00",
+                    "status": 400,
+                    "msg": "Authentication is required"
+                }
+            }
+        )
+    }
+)
+
+@api_view(['GET'])
+def get_message_count_by_bot(request):
+    # Check the request header for email
+    if not request.headers or HEADER_USER_EMAIL not in request.headers:
+        return generate_error_response(400, "Authentication is required")
+
+    email = request.headers.get(HEADER_USER_EMAIL)
+    if not email:
+        return generate_error_response(400, "Authentication is required")
+
+    # Get the bot's ID from the request parameters
+    bot_id = request.GET.get('botId', None)
+    if bot_id is None:
+        return generate_error_response(400, "Bot ID is required")
+
+    # Check if the bot ID is a valid UUID
+    try:
+        uuid.UUID(bot_id)
+    except ValueError:
+        return generate_error_response(400, "Invalid Bot ID")
+
+    # Get sorting order from URL parameter
+    sorting_order = request.GET.get('sort', 'desc')
+
+    # Get the value of num_days from the query parameters
+    num_days = request.GET.get('daysLimit', None)
+    if num_days is not None:
+        try:
+            num_days = int(num_days)
+        except ValueError:
+            return generate_error_response(400, "Invalid value for 'num_days'. Must be an integer.")
+
+    # Calculate the start date based on num_days
+    if num_days is not None:
```

Comment:
Between lines `92` and `99`, `num_days is not None` is checked twice. It seems possible to merge the two checks into one, which would be better for performance and readability of the code

Reply:
Merged the two checks